jvm/src/main/java/com/microsoft/bond/protocol/StreamHelper.java

```diff
@@ -0,0 +1,45 @@
+package com.microsoft.bond.protocol;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+/**
+ * Contains helper methods for working with input and output streams.
+ */
+final class StreamHelper {
+
+    /**
+     * Reads a single byte from the input, raising an EOFException if end of stream is reached.
+     * @param inputStream source input stream
+     * @return a byte
+     * @throws EOFException if end of stream reached
+     * @throws IOException if other I/O error occurred
+     */
+    static byte readByte(InputStream inputStream) throws IOException {
+        int b = inputStream.read();
+        if (b < 0) {
+            // EOFException is self-explanatory and doesn't need a message
```

Comment:
I added it for completeness. Per other comment, I had to document EOFException but the method's declaration specifies it throws IOException, hence documented that just for completeness.

---
In reply to: [116105899](https://github.com/tstein/bond/pull/4#discussion_r116105899) [](ancestors = 116105899)